### ISSUE TITLE
How can we make shell integration smarts more discoverable to screen reader users?

### ISSUE THREAD

**Issue by Tyriar** [microsoft/vscode#170634]
Shell integration gives go to recent directory and run recent command which are very useful when using a screen reader. How can we make these more discoverable in product when using a screen reader?

**Comment by meganrogge** [comment]
@isidorn if you have any ideas here or prior art

**Comment by isidorn** [comment]
Can you provide a bit more context for me here please :)
Can we make them more discoverable for all users? Why do we think this is particularly helpful for screen reader users?

**Comment by Tyriar** [comment]
AFAIK ctrl+r to search command history generally isn't used by screen reader users as it's difficult to understand what gets announced:

<img width="199" alt="image" src="https://user-images.githubusercontent.com/2193314/211026605-f02c2483-d2f5-4e9d-9e77-8a61a9ac5cf8.png">

This gives an accessible alternative with additional features:

<img width="450" alt="image" src="https://user-images.githubusercontent.com/2193314/211026639-331f3d2d-719f-47ba-95b3-0f2515648d5d.png">

Go to recent directory is similar but there isn't typically an equivalent in most shells:

<img width="453" alt="image" src="https://user-images.githubusercontent.com/2193314/211026746-9f69a156-4085-40fe-b8ce-586fab173183.png">

The issue is how does the user discover that this enhanced functionality exists without intercepting ctrl+r/g and getting them to choose via a notification, which is a crappy experience for users who don't want it. Something I really want to ensure is maintaining a great experience for "pro" users who have invested in configuring their shell just how they like it.

One idea I had the other day was when accessibility mode is on, add these as default keybindings. That has issues too though; what if the user still wants ctrl+r/ctrl+g to go to the shell, we didn't educate them on how to do that, plus it doesn't solve the discoverability problem when accessibility mode is not enabled (it's a cool feature for most users).

**Comment by isidorn** [comment]
Thanks for providing more context. I like the default shortcut idea when accessibility mode is on. For context: we also do this for editor word navigation shortcuts, we change the navigation commands so they are aligned with what screen reader expects from Chromium word navigation.

Apart from that then we only need to think about the general discoverability problem. Did we consider having a top level `...` title entry where we could put some terminal commands so users discover them? Alternative: terminal context menu submenu?

**Comment by Tyriar** [comment]
We did want to add additional buttons to the view title which are hidden by default so maybe `...` is the answer. Right click for context menu doesn't work for on all OS's so that's not the best option.

**Comment by Tyriar** [comment]
Hidden by default menu items that can be toggled: https://github.com/microsoft/vscode/pull/170727

**Comment by Tyriar** [comment]
Another idea: add a new section in the middle which is global shell integration commands (go to recent/run recent command) and also change configure to configure shell integration which allows toggling ctrl+r (and ctrl+g keybindings?\*). By default we could set ctrl+alt+r to be native run recent command.

<img width="263" alt="image" src="https://user-images.githubusercontent.com/2193314/211370748-438dd60e-a44c-41f3-85e0-3587bd5640f2.png">

\* Should ctrl+g be the default?

**Comment by isidorn** [comment]
I like the `Alt+F1` idea.
Not sure about that section in the middle - I would like to try it out to provide quality feedback.

Unrelated topic: are we sure "shell integration" is a good and easy to understand name?

**Comment by Tyriar** [comment]
Possible name confusion has come up before but trying to come up with a new name for this pretty common terminal feature at this point would probably be worse imo.

**Comment by Tyriar** [comment]
I forked off the ideas from this into #170976, #170977 and #170979 so any further discussion is more focused and to ensure we'll verify and mention in release notes.
